packages/labs/react/src/create-component.ts

```diff
@@ -6,6 +6,20 @@
 
 import * as ReactModule from 'react';
 
+type Constructor<I> = {new (): I};
+type CustomEventListener = (e: CustomEvent) => void;
+type EventListeners = EventListener | CustomEventListener;
+type Events<R> = {
+  [K in keyof R]?: EventListeners;
+};
+
+type ReservedReactProperties =
+  | 'children'
+  | 'localName'
+  | 'ref'
+  | 'style'
+  | 'className';
+
```

Comment:
To avoid repeating the property names you can infer the type from a const array: https://www.typescriptlang.org/play?#code/MYewdgzgLgBATgUwguA3BATASgghsKABThAAcUoBPAOVwFskYBeGAbQCgYYByYACwCWAGwyIw3ADSceQkMFxDaDSdO6IAZiq7dolIQi09gQ3BAhKDUgLoxTMUJCgBudg+jwkKdNjwFiZCgFGFjAEAHcYAGUEKAAKAEoXdipyGABRfQYwKAAVSnIAHhyYBAAPKAQwDAgYHFwMcCFKAEE4OFxKAoBXMABrMBAwsAA+YeZpYrKKqpq6hrAm1vbOgTB1FHTMytz8hGHpLgB+TYQsnfIDmAAuGFD0OCSUhFrPNEw6vxJyOCoLZhOznlCk8QOoPMg3j58EQvhQaPQkPtXOB3KQvhYbjgId4PjCAj94Qx-tx1CAQNwnEA

Reply:
i'll try!

Reply:
done!